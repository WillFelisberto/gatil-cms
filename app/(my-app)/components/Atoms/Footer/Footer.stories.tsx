import type { Meta, StoryObj } from '@storybook/react';
import { Facebook, Instagram, Music2 } from 'lucide-react';

import { Footer } from './Footer';

const meta: Meta<typeof Footer> = {
  title: 'Atoms/Footer',
  component: Footer,
  tags: ['autodocs'],
  parameters: {
    layout: 'fullscreen'
  },
  decorators: [
    (Story) => (
      <div className="min-h-screen flex flex-col">
        {/* Conteúdo da página (simulado) */}
        <main className="flex-grow p-6 bg-gray-50">
          <p className="text-center text-gray-400">
            Conteúdo da página simulada para visualização do Footer no rodapé.
          </p>
        </main>
        <Story />
      </div>
    )
  ]
};

export default meta;
type Story = StoryObj<typeof Footer>;

export const Default: Story = {
  args: {
    socialLinks: [
      { icon: Facebook, url: 'https://facebook.com' },
      { icon: Instagram, url: 'https://instagram.com' },
      { icon: Music2, url: 'https://tiktok.com' }
    ],
    menuItems: [
      { label: 'Início', href: '#' },
      { label: 'Apadrinhe', href: '#' },
      { label: 'Colabore', href: '#' },
      { label: 'Contato', href: '#' },
      { label: 'Sobre o Projeto', href: '#' },
      { label: 'Adote', href: '#' }
    ]
  }
};
