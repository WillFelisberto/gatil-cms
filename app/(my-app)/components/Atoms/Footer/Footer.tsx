'use client';

import { Facebook, Instagram, Music2 } from 'lucide-react';
import Image from 'next/image';
import { usePathname } from 'next/navigation';

const iconMap = {
  facebook: Facebook,
  instagram: Instagram,
  tiktok: Music2
};
type SocialLink = {
  type: 'facebook' | 'instagram' | 'tiktok';
  url: string;
};

type MenuItem = {
  label: string;
  href: string;
};

type FooterProps = {
  socialLinks: SocialLink[];
  menuItems: MenuItem[];
};

export const Footer = ({ socialLinks, menuItems }: FooterProps) => {
  const pathname = usePathname();

  return (
    <footer
      className="flex flex-col border-t border-gray-200 bg-white"
      role="contentinfo"
      data-testid="footer"
    >
      <div className="w-full flex justify-center">
        <div
          className="w-full max-w-7xl px-6 py-10 flex flex-col md:flex-row justify-between items-center md:items-start text-center md:text-left gap-10 md:gap-20"
          data-testid="footer-container"
        >
          {/* Redes sociais e texto */}
          <div className="flex-1 justify-center" data-testid="footer-social">
            <div className="flex justify-center mb-4">
              <Image
                src="/logo-rodape.webp"
                alt="Logo do Gatil dos Resgatados"
                width={80}
                height={80}
                className="object-contain"
              />
            </div>
            <p className="text-base text-gray-700 max-w-md mx-auto text-center font-poppins">
              Siga-nos nas redes sociais e acompanhe de perto as histórias dos nossos gatinhos
              resgatados, eventos e novidades! Junte-se à nossa comunidade e ajude a espalhar essa
              causa
            </p>
            <div
              className="flex justify-center gap-4 mt-4 text-blue-950 text-xl"
              aria-label="Redes sociais"
            >
              {socialLinks.map(({ type, url }, index) => {
                const Icon = iconMap[type];

                return (
                  <a
                    href={url}
                    key={index}
                    target="_blank"
                    rel="noopener noreferrer"
                    className="hover:text-blue-600 hover:scale-110 active:text-blue-700 transition-colors duration-200"
                    data-testid={`footer-social-link-${index}`}
                  >
                    <Icon size={20} />
                  </a>
                );
              })}
            </div>
          </div>

          {/* Menu Rápido */}
          <nav
            className="flex-1 justify-center text-center"
            aria-label="Menu Rápido"
            data-testid="footer-menu"
          >
            <h4 className="font-semibold text-base mb-4 text-blue-950">Menu Rápido</h4>
            <ul className="space-y-2 text-sm text-gray-800">
              {menuItems.map(({ label, href }, index) => {
                const isActive = pathname === href;

                return (
                  <li key={index}>
                    <a
                      href={href}
                      data-testid={`footer-menu-link-${index}`}
                      className={`transition-colors duration-200 hover:text-blue-600 hover:font-bold ${
                        isActive ? 'text-primary-blue font-semibold underline' : ''
                      }`}
                    >
                      {label}
                    </a>
                  </li>
                );
              })}
            </ul>
          </nav>
        </div>
      </div>

      <div className="bg-blue-950 text-white text-center py-4 w-full text-sm">
        Feito com 😻 por Gatil dos Resgatados
      </div>
    </footer>
  );
};
