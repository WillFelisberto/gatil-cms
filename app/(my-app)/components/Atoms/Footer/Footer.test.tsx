import { render, screen } from '@/tests/test-utils';

import { Footer } from './Footer';

jest.mock('next/navigation', () => ({
  usePathname: jest.fn()
}));

import { usePathname } from 'next/navigation';

describe('<Footer />', () => {
  const mockSocialLinks: { type: 'facebook' | 'instagram' | 'tiktok'; url: string }[] = [
    { type: 'facebook', url: 'https://facebook.com' },
    { type: 'instagram', url: 'https://instagram.com' },
    { type: 'tiktok', url: 'https://tiktok.com' }
  ];

  const mockMenuItems = [
    { label: 'InÃ­cio', href: '/' },
    { label: 'Apadrinhe', href: '/apadrinhe' },
    { label: 'Contato', href: '/contato' }
  ];

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render the footer container', () => {
    (usePathname as jest.Mock).mockReturnValue('/');
    const { container } = render(
      <Footer socialLinks={mockSocialLinks} menuItems={mockMenuItems} />
    );
    expect(screen.getByTestId('footer')).toBeInTheDocument();
    expect(screen.getByTestId('footer-container')).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  it('should render all social media icons with correct hrefs', () => {
    (usePathname as jest.Mock).mockReturnValue('/');
    render(<Footer socialLinks={mockSocialLinks} menuItems={mockMenuItems} />);
    mockSocialLinks.forEach((_, index) => {
      const link = screen.getByTestId(`footer-social-link-${index}`);
      expect(link).toBeInTheDocument();
      expect(link).toHaveAttribute('href', mockSocialLinks[index].url);
      expect(link).toHaveAttribute('rel', 'noopener noreferrer');
      expect(link).toHaveAttribute('target', '_blank');
    });
  });

  it('should render all menu items with correct labels and hrefs', () => {
    (usePathname as jest.Mock).mockReturnValue('/');
    render(<Footer socialLinks={mockSocialLinks} menuItems={mockMenuItems} />);
    mockMenuItems.forEach((item, index) => {
      const link = screen.getByTestId(`footer-menu-link-${index}`);
      expect(link).toBeInTheDocument();
      expect(link).toHaveTextContent(item.label);
      expect(link).toHaveAttribute('href', item.href);
    });
  });

  it('should mark the current page as active using pathname', () => {
    (usePathname as jest.Mock).mockReturnValue('/apadrinhe');
    render(<Footer socialLinks={mockSocialLinks} menuItems={mockMenuItems} />);
    const activeLink = screen.getByTestId('footer-menu-link-1');
    expect(activeLink).toHaveClass('text-primary-blue');
    expect(activeLink).toHaveClass('font-semibold');
    expect(activeLink).toHaveClass('underline');
  });

  it('should render default copyright', () => {
    (usePathname as jest.Mock).mockReturnValue('/');
    render(<Footer socialLinks={mockSocialLinks} menuItems={mockMenuItems} />);
    expect(screen.getByText(/feito com ðŸ˜» por/i)).toBeInTheDocument();
  });

  it('should render correctly even with empty props', () => {
    (usePathname as jest.Mock).mockReturnValue('/');
    render(<Footer socialLinks={[]} menuItems={[]} />);
    expect(screen.getByTestId('footer')).toBeInTheDocument();
    expect(screen.queryByTestId('footer-social-link-0')).not.toBeInTheDocument();
    expect(screen.queryByTestId('footer-menu-link-0')).not.toBeInTheDocument();
  });
});
