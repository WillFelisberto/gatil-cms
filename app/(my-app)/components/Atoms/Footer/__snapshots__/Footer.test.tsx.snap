// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render the footer container 1`] = `
<div>
  <footer
    class="flex flex-col border-t border-gray-200 bg-white"
    data-testid="footer"
    role="contentinfo"
  >
    <div
      class="w-full flex justify-center"
    >
      <div
        class="w-full max-w-7xl px-6 py-10 flex flex-col md:flex-row justify-between items-center md:items-start text-center md:text-left gap-10 md:gap-20"
        data-testid="footer-container"
      >
        <div
          class="flex-1 justify-center"
          data-testid="footer-social"
        >
          <div
            class="flex justify-center mb-4"
          >
            <img
              alt="Logo do Gatil dos Resgatados"
              class="w-20 h-20 object-contain"
              src="/logo-rodape.webp"
            />
          </div>
          <p
            class="text-base text-gray-700 max-w-md mx-auto text-center font-poppins"
          >
            Siga-nos nas redes sociais e acompanhe de perto as histÃ³rias dos nossos gatinhos resgatados, eventos e novidades! Junte-se Ã  nossa comunidade e ajude a espalhar essa causa
          </p>
          <div
            aria-label="Redes sociais"
            class="flex justify-center gap-4 mt-4 text-blue-950 text-xl"
          >
            <a
              class="hover:text-blue-600 hover:scale-110 active:text-blue-700 transition-colors duration-200"
              data-testid="footer-social-link-0"
              href="https://facebook.com"
              rel="noopener noreferrer"
              target="_blank"
            >
              <svg
                class="lucide lucide-facebook"
                fill="none"
                height="20"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M18 2h-3a5 5 0 0 0-5 5v3H7v4h3v8h4v-8h3l1-4h-4V7a1 1 0 0 1 1-1h3z"
                />
              </svg>
            </a>
            <a
              class="hover:text-blue-600 hover:scale-110 active:text-blue-700 transition-colors duration-200"
              data-testid="footer-social-link-1"
              href="https://instagram.com"
              rel="noopener noreferrer"
              target="_blank"
            >
              <svg
                class="lucide lucide-instagram"
                fill="none"
                height="20"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  height="20"
                  rx="5"
                  ry="5"
                  width="20"
                  x="2"
                  y="2"
                />
                <path
                  d="M16 11.37A4 4 0 1 1 12.63 8 4 4 0 0 1 16 11.37z"
                />
                <line
                  x1="17.5"
                  x2="17.51"
                  y1="6.5"
                  y2="6.5"
                />
              </svg>
            </a>
            <a
              class="hover:text-blue-600 hover:scale-110 active:text-blue-700 transition-colors duration-200"
              data-testid="footer-social-link-2"
              href="https://tiktok.com"
              rel="noopener noreferrer"
              target="_blank"
            >
              <svg
                class="lucide lucide-music2 lucide-music-2"
                fill="none"
                height="20"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="8"
                  cy="18"
                  r="4"
                />
                <path
                  d="M12 18V2l7 4"
                />
              </svg>
            </a>
          </div>
        </div>
        <nav
          aria-label="Menu RÃ¡pido"
          class="flex-1 justify-center text-center"
          data-testid="footer-menu"
        >
          <h4
            class="font-semibold text-base mb-4 text-blue-950"
          >
            Menu RÃ¡pido
          </h4>
          <ul
            class="space-y-2 text-sm text-gray-800"
          >
            <li>
              <a
                class="transition-colors duration-200 hover:text-blue-600 hover:font-bold text-primary-blue font-semibold underline"
                data-testid="footer-menu-link-0"
                href="/"
              >
                InÃ­cio
              </a>
            </li>
            <li>
              <a
                class="transition-colors duration-200 hover:text-blue-600 hover:font-bold "
                data-testid="footer-menu-link-1"
                href="/apadrinhe"
              >
                Apadrinhe
              </a>
            </li>
            <li>
              <a
                class="transition-colors duration-200 hover:text-blue-600 hover:font-bold "
                data-testid="footer-menu-link-2"
                href="/contato"
              >
                Contato
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
    <div
      class="bg-blue-950 text-white text-center py-4 w-full text-sm"
    >
      Feito com ðŸ˜» por Gatil dos Resgatados
    </div>
  </footer>
</div>
`;
